src/comment.rs

```diff
@@ -647,23 +644,28 @@ impl<'a> CommentRewrite<'a> {
         } else if self.code_block_attr.is_some() {
             if line.starts_with("```") {
                 let code_block = match self.code_block_attr.as_ref().unwrap() {
-                    CodeBlockAttribute::Ignore | CodeBlockAttribute::Text => {
+                    CodeBlockAttribute::NotRust => {
                         trim_custom_comment_prefix(&self.code_block_buffer)
                     }
-                    _ if self.code_block_buffer.is_empty() => String::new(),
-                    _ => {
-                        let mut config = self.fmt.config.clone();
-                        config.set().wrap_comments(false);
-                        if config.format_code_in_doc_comments() {
-                            if let Some(s) =
-                                crate::format_code_block(&self.code_block_buffer, &config, false)
-                            {
-                                trim_custom_comment_prefix(&s.snippet)
+                    CodeBlockAttribute::Rust => {
+                        if self.code_block_buffer.is_empty() {
+                            String::new()
+                        } else {
+                            let mut config = self.fmt.config.clone();
+                            config.set().wrap_comments(false);
+                            if config.format_code_in_doc_comments() {
+                                if let Some(s) = crate::format_code_block(
+                                    &self.code_block_buffer,
+                                    &config,
+                                    false,
+                                ) {
+                                    trim_custom_comment_prefix(&s.snippet)
+                                } else {
+                                    trim_custom_comment_prefix(&self.code_block_buffer)
+                                }
                             } else {
                                 trim_custom_comment_prefix(&self.code_block_buffer)
                             }
-                        } else {
-                            trim_custom_comment_prefix(&self.code_block_buffer)
                         }
                     }
```

Comment:
Overall changes look good to me. Somewhat of a nit, but I think if we were to move some of the conditionals up as guards on arms then we could reduce the nested conditional complexity and accompanying indentation levels, example below. What do you think?

```rust
CodeBlockAttribute::Rust if self.code_block_buffer.is_empty() => String::new(),
CodeBlockAttribute::Rust if config.format_code_in_doc_comments() => {
    let mut config = self.fmt.config.clone();
    // ....
}
CodeBlockAttribute::Rust => trim_custom_comment_prefix(&self.code_block_buffer),